pkg/provider/azure_controller_common.go

```diff
@@ -147,8 +147,20 @@ func (c *controllerCommon) getNodeVMSet(nodeName types.NodeName, crt azcache.Azu
 // return (lun, error)
 func (c *controllerCommon) AttachDisk(ctx context.Context, async bool, diskName, diskURI string, nodeName types.NodeName,
 	cachingMode compute.CachingTypes, disk *compute.Disk) (int32, error) {
+	// lun channel is used to return assigned lun values preemptively
+	var lunCh chan int32
```

Comment:
Consider moving the `lunCh` declaration and intialization to after the `waitForBatch` declaration so that you only need defer the `close` if we actually get a channel.